internal/fleet/agent_policy_resource.go

```diff
@@ -264,6 +345,25 @@ func resourceAgentPolicyRead(ctx context.Context, d *schema.ResourceData, meta i
 		}
 	}
 
+	if agentPolicy.GlobalDataTags != nil {
+		if len(agentPolicy.GlobalDataTags) > 0 && serverVersion.LessThan(minVersionGlobalDataTags) {
+			return diag.FromErr(fmt.Errorf("'global_data_tags' is supported only for Elasticsearch v%s and above", minVersionGlobalDataTags.String()))
+		}
```

Comment:
```suggestion
```

We can probably skip the version check in the read scenario. It seems reasonable to assume that if the API returns the global tags then they're supported?

Reply:
sounds good to me